api/src/main/java/ai/djl/nn/norm/LayerNorm.java

```diff
@@ -0,0 +1,256 @@
+ * and limitations under the License.
+ */
+package ai.djl.nn.norm;
+
+import ai.djl.Device;
+import ai.djl.MalformedModelException;
+import ai.djl.ndarray.NDArray;
+import ai.djl.ndarray.NDList;
+import ai.djl.ndarray.internal.NDArrayEx;
+import ai.djl.ndarray.types.Shape;
+import ai.djl.nn.AbstractBlock;
+import ai.djl.nn.Parameter;
+import ai.djl.training.ParameterStore;
+import ai.djl.util.PairList;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * Layer normalization works by normalizing the values of input data for each input sample to have
+ * mean of 0 and variance of 1. Since this may alter the representation of a layer, two parameters
+ * (\ (\gamma\) and \(\beta\)) are learned along the normalization process to respectively scale and
+ * shift the normalized output (activations) to have any mean and variance so the network can
+ * utilize non-linear transformations such as sigmoid function as described in the <a
+ * href="https://arxiv.org/abs/1607.06450">paper</a>. During backpropagation, both \(\gamma\) and
+ * \(\beta\) parameters are included following the chain-rule in derivation.
+ *
+ * <p>Citing the abstract of the paper: "Training state-of-the-art, deep neural networks is
+ * computationally expensive. One way to reduce the training time is to normalize the activities of
+ * the neurons. A recently introduced technique called batch normalization uses the distribution of
+ * the summed input to a neuron over a mini-batch of training cases to compute a mean and variance
+ * which are then used to normalize the summed input to that neuron on each training case. This
+ * significantly reduces the training time in feed-forward neural networks. However, the effect of
+ * batch normalization is dependent on the mini-batch size and it is not obvious how to apply it to
+ * recurrent neural networks. In this paper, we transpose batch normalization into layer
+ * normalization by computing the mean and variance used for normalization from all of the summed
+ * inputs to the neurons in a layer on a single training case. Like batch normalization, we also
+ * give each neuron its own adaptive bias and gain which are applied after the normalization but
+ * before the non-linearity. Unlike batch normalization, layer normalization performs exactly the
+ * same computation at training and test times. It is also straightforward to apply to recurrent
+ * neural networks by computing the normalization statistics separately at each time step. Layer
+ * normalization is very effective at stabilizing the hidden state dynamics in recurrent networks.
+ * Empirically, we show that layer normalization can substantially reduce the training time compared
+ * with previously published techniques."
+ */
+public class LayerNorm extends AbstractBlock {
+
+    private static final byte VERSION = 2;
```

Comment:
We can start from Version 1, this is used for serialize and deserialize the block

Reply:
fixed